### ISSUE TITLE
Introduce testing suite

### ISSUE THREAD

**Issue by maxpert** [maxpert/marmot#27]
Following are required to be done:
 - Integration tests on CDC watching and publishing to NATS
 - Integration tests of reapplying change logs from NATS
 - Snapshot save/restore testing
 - SQLite Tcl level testing suite integration

**Comment by gedw99** [comment]
yep agree that development needs a system to consistently see bugs.

also devs that experience bugs can supply seed data for reproduction of bug, so that we can both track down bugs but add them into seed data and tests to avoid regressions. 

Are there any proposals for how to do this ?

- seed data into SQL
- Tests that work against the seed data 
- Nats cluster setup and tear down 
- Chaos on nats Cluster like loosing one or upgrading using lame duck mode, so we related failure modes.

Do we presume Linus only ? Some users might be running SQLite on desktops with marmot on Linus server.
